src/TextInput.jsx

```diff
@@ -75,7 +81,8 @@ TextInput.propTypes = {
 		React.PropTypes.element
 	]),
 	labelClassName: React.PropTypes.string,
-	required: React.PropTypes.bool
+	required: React.PropTypes.bool,
+	onChange: React.PropTypes.func.isRequired,
```

Comment:
this feels like it should be optional - does a parent have to do something every time the value changes?